src/main/java/com/javatemplate/domain/book/BookService.java

```diff
@@ -53,17 +67,26 @@ public Book update(final UUID bookId, final Book bookUpdate) {
 
     public void deleteById(final UUID uuid) {
         final Book book = findById(uuid);
+        validateBookPermissions(book.getUserId(), "delete");
 
         bookStore.deleteById(book.getId());
     }
 
+
+    private void validateBookPermissions(final UUID userId, final String action) {
```

Comment:
Phải có sự hoà hợp nhất định giữa tên hàm và các tham số. Ở đây tên hàm khi đọc vào sẽ kiểu "kiểm tra quyền hạn của sách", hoàn toàn không hiểu userId ở đây là cái gì, action là cái gì. Validate quyền 1 cuốn sách thì tại sao không hề có tham số gì đó về sách và quyền hạn gì.


```suggestion
    private void validateDeletePermission(final Book book) {
```

Reply:
E hiểu rồi anh. Added